api/src/main/java/org/eclipse/microprofile/concurrent/ManagedExecutorConfig.java

```diff
@@ -19,27 +19,39 @@
 package org.eclipse.microprofile.concurrent;
 
 import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
 /**
- * <p>Configures an injected <code>ManagedExecutor</code> instance.</p>
+ * <p>Provides configuration for an injected {@link ManagedExecutor} instance
+ * which is {@link javax.enterprise.context.Dependent dependent} scoped.
+ * When an application has multiple injection points for {@link ManagedExecutor}
+ * with matching configuration, the container must inject the same instance.
+ * For the purposes of matching, array attributes of this annotation are
+ * considered as unordered sets where duplicate elements are ignored.</p>
  *
  * <p>Example usage:</p>
  * <pre><code> &commat;Inject &commat;ManagedExecutorConfig(propagated=ThreadContext.CDI, maxAsync=5)
  * ManagedExecutor executor;
  * ...
  * </code></pre>
  *
- * <p>A <code>ManagedExecutor</code> must fail to inject, raising
+ * <p>A {@link ManagedExecutor}'s life cycle is tied to that of the application.
```

Comment:
Since it is a CDI bean with dependent scope, the lifecycle of such bean is tied to the bean into which it is injected. 
E.g. if you inject it into you `Foo` bean and the context in which `Foo` sits is destroyed, then CDI attempts to destroy this dependent bean as well (calling PreDestroy/Dispose methods). Now this is effectively overriden by the enforced sharing as you keep a reference to that instance hence it stays alive. This isn't a problem but the sentence is somewhat misleading.

How about something like this: `All created instances of ManagedExecutor are destroyed when the application is stopped and the container then automatically shuts down all managed executors and cancels its remaining actions/tasks`

Reply:
I like your proposed changing in wording. I'll switch it to that.